### PR TITLE
#231 - LineSegment

### LINKED ISSUE #231: add LineSegment type

### PR DESCRIPTION
**Pull request by schillic**
Closes #231.

* [x] type + support vector
* [x] containment check
* [x] intersection emptiness check
* [x] subset check (uses `AbstractConvexSet`, hence commented out)
* [x] plot recipe

### CODE REVIEW

File: src/LineSegment.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- `d` -- direction
+- `L` -- line segment
+
+### Output
+
+The support vector in the given direction.
+
+### Algorithm
+
+If the angle between the vector ``q - p`` and ``d`` is bigger than 90° and less
```
Comment (mforets) [inline src/LineSegment.jl]: add "... measured in counter-clockwise order"

**Comment by mforets** [comment]
This can be a plotting function:

```julia
@recipe function plot_LineSegment(l::LineSegment; color="blue", label="",
                                 grid=true, alpha=0.5, legend=false, add_marker=true)

    seriestype := :path
    linecolor   --> "blue"
    markershape --> (add_marker ? :circle : :none)
    [Tuple(l.p); Tuple(l.q)]
end
```

**Comment by schillic** [comment]
Do you want to add the plot recipe or should I do it?

File: src/LineSegment.jl [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Type that represents a line segment in 2D between two points ``p`` and ``q``.
+
+### Fields
+
+- `p` -- first point
+- `q` -- second point
+"""
+struct LineSegment{N<:Real} <: LazySet{N}
+    p::Vector{N}
```
Comment (mforets) [inline src/LineSegment.jl]: this is a case where static vectors can be useful if heavy use (like arithmetic) over line segments is performed => should we make it an `AbstractVector`?

Comment (schillic) [reply src/LineSegment.jl]: okay :+1:

**Comment by mforets** [comment]
> Do you want to add the plot recipe or should I do it?

i can add that

File: src/is_intersection_empty.jl [hunk 3]
```diff
@@ -418,3 +418,130 @@ function is_intersection_empty(H::Hyperplane{N},
...
+We first check if the two line segments are parallel, and if so, if they are
+collinear.
+In the latter case, we check containment of any of the end points in the other
+line segment.
+Otherwise the lines are not parallel, so we can solve an equation of the
+intersection point, if it exists.
+
+### Notes
+
+Witness production is currently not supported.
```
Comment (mforets) [inline src/is_intersection_empty.jl]: > Witness production is currently not supported.

but a witness is returned in some lines below. copy paste typo?

Comment (schillic) [reply src/is_intersection_empty.jl]: Oh, I forgot to remove that. Thanks, I'll fix it now.

**Comment by schillic** [comment]
I added the plot recipes --> merge?

**Pull request merged**